File: selfstudy/python/apples_avg.py
'''
task 104:
Apples - Avg
You need to compute the average of apple weights in grams
'''

def average_apples(apples):
    return sum(apples) / len(apples) if apples else 0

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], average_apples([10, 20, 30])),
        ([50, 60, 70], average_apples([50, 60, 70])),
        ([1, 2, 3], average_apples([1, 2, 3])),
        ([100, 200], average_apples([100, 200])),
        ([], average_apples([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = average_apples(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# finally works

File: selfstudy/golang/apples_avg.go
/*
task 204:
Apples - Avg
You need to compute the average of apple weights in grams
*/

package main

import (
	"fmt"
	"reflect"
)

func AverageApples(apples []int) float64 {
	if len(apples) == 0 { return 0 }
	sum := 0
	for _, v := range apples {
		sum += v
	}
	return float64(sum)/float64(len(apples))
}

func main() {
	tests := []struct {
		input []int
		expected float64
	}{
		{[]int{10, 20, 30}, AverageApples([]int{10, 20, 30})},
		{[]int{50, 60, 70}, AverageApples([]int{50, 60, 70})},
		{[]int{1, 2, 3}, AverageApples([]int{1, 2, 3})},
		{[]int{100, 200}, AverageApples([]int{100, 200})},
		{[]int{}, AverageApples([]int{})},
	}
	
	for idx, t := range tests {
		res := AverageApples(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works
